improv_ble_client/protocol.py

```diff
@@ -300,41 +301,72 @@ class DeviceInfoRes(Command):
     firmware_version: bytes
     hardware_chip: bytes
     device_name: bytes
+    os_name: bytes | None
+    os_version: bytes | None
 
     def __init__(
         self,
         firmware_name: bytes,
         firmware_version: bytes,
         hardware_chip: bytes,
         device_name: bytes,
+        os_name: bytes | None = None,
+        os_version: bytes | None = None,
     ) -> None:
         """Initialize."""
-        super().__init__([firmware_name, firmware_version, hardware_chip, device_name])
+        super().__init__(
+            [firmware_name, firmware_version, hardware_chip, device_name]
+            if os_name is None
+            else [
+                firmware_name,
+                firmware_version,
+                hardware_chip,
+                device_name,
+                os_name,
+                os_version,
+            ]
+        )
         self.firmware_name = firmware_name
         self.firmware_version = firmware_version
         self.hardware_chip = hardware_chip
         self.device_name = device_name
+        self.os_name = os_name
+        self.os_version = os_version
 
     def __str__(self) -> str:
         return (
             f"{self.__class__.__name__} firmware:{self.firmware_name.decode()}, "
             f"version:{self.firmware_version.decode()}, "
             f"chip:{self.hardware_chip.decode()}, "
             f"name:{self.device_name.decode()}"
+            f", os:{self.os_name.decode()}"
+            if self.os_name is not None
+            else (
+                "" f", os_version:{self.os_version.decode()}"
+                if self.os_version is not None
+                else ""
+            )
         )
 
     @classmethod
     def from_bytes(cls, data: bytes) -> DeviceInfoRes:
         """Initialize from serialized representation of the command."""
         cls._validate(data)
         strings = cls._extract_strings(data)
-        return cls(strings[0], strings[1], strings[2], strings[3])
+        return cls(
+            strings[0],
+            strings[1],
+            strings[2],
+            strings[3],
+            strings[4] if len(strings) > 4 else None,
+            strings[5] if len(strings) > 5 else None,
+        )
 
     @classmethod
     def _validate(cls, data: bytes) -> None:
         """Raise if the data is not valid."""
         super()._validate(data)
-        if len(cls._extract_strings(data)) != 4:
```

Comment:
This was technically a bug before as sending more than 4 should not have caused an error.